### PR TITLE
Admin features frontend

### PR DESCRIPTION
**Pull request by Brunettow**
Since some of the features implemented in this branch is needed by others, It will be merged to the frontend.

So far admin feature buttons are implemented.
Visibility of the buttons are adjusted however not connected to the backend.
Only basic user provider is used in profile page to get user type.

### CODE REVIEW

**Comment by laylaylo** [comment]
Now, without even login, I can see admin buttons on node page. Is this intended implementation? 
If user type provider is connected and getting the type, why it is not used? Is it currently only for basic user, I didn't understand that part totally.

**Review by laylaylo — CHANGES_REQUESTED** [review]
I think there is a **bug** about seeing profile page in this branch, because it works fine on frontend branch. It's not possible to visit profile pages as a guest (without logging).  Can you please check it out?

File: project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/profile_page.dart [hunk 1]
```diff
@@ -101,238 +113,292 @@ class _ProfilePageState extends State<ProfilePage> {
...
+      isReviewer = !isReviewer; // Toggle the state for example purposes
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final User? user = Provider.of<Auth>(context).user;
     if (user == null) {
-      // guest can see profile pages
+      // TODO guest can see profile pages
```
Comment (laylaylo) [inline project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/profile_page.dart:151]: oh I see that it's not a bug but a whole logic change here, need to understand why this is all changed. Could you please briefly summarize?

Comment (laylaylo) [reply project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/profile_page.dart:151]: After a discussion, we realized that there was a misunderstanding about these comments. Please delete TODO parts.

**Comment by Brunettow** [comment]
Thank you for catching this bug. I fixed and re-uploaded the code.
Now, guest user can see the profile pages without an error.

File: project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/about_me.dart [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
 import 'package:flutter/material.dart';
 
-class AboutMe extends StatelessWidget {
+class AboutMe extends StatefulWidget {
   final String email;
   final String name;
   final String surname;
   final int noWorks;
   final String aboutMe;
+  final bool isBanned;
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/about_me.dart:13]: woah pretty complicated and scary constructor. There are always better ways to pass these kind of informations. Just keep in mind not to do that in the future, but for this class as it is used by just one parent widget, that's OK

File: project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/about_me.dart [hunk 3]
```diff
@@ -27,15 +46,87 @@ class AboutMe extends StatelessWidget {
...
                     color: AppColors.primaryDarkColor,
                     fontWeight: FontWeight.bold,
                     fontSize: 40,
                   ),
                 ),
+                Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.end,
+                  children: [
+                    Visibility(
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/about_me.dart:63]: Nice widget use! Visibility widget made everything clear.

**Comment by Brunettow** [comment]
Buttons on node view page are not connected to the backend yet. That's why you can see it unlike profile page. It will be implemented in the next PR. Since some of the implementations are needed by @Zulalm, we need this branch to be merged to the frontend branch.

**Review by laylaylo — APPROVED** [review]
nice work

**Pull request merged**